src/spatialdata_plot/pl/utils.py

```diff
@@ -1886,20 +1945,47 @@ def _get_wanted_render_elements(
     raise ValueError(f"Unknown element type {element_type}")
 
 
-def _update_params(sdata, params, wanted_elements_on_cs, element_type: Literal["images", "labels", "points", "shapes"]):
-    if element_type in ["labels", "points", "shapes"] and wanted_elements_on_cs:
+def _update_params(
+    sdata: SpatialData,
+    params: ImageRenderParams | LabelsRenderParams | PointsRenderParams | ShapesRenderParams,
+    wanted_elements_on_cs: list[str],
+    element_type: Literal["images", "labels", "points", "shapes"],
+) -> ImageRenderParams | LabelsRenderParams | PointsRenderParams | ShapesRenderParams:
+    if isinstance(params, (LabelsRenderParams, PointsRenderParams, ShapesRenderParams)) and wanted_elements_on_cs:
         params = _create_initial_element_table_mapping(sdata, params, wanted_elements_on_cs)
-        if element_type == "labels":
+        if isinstance(params, LabelsRenderParams):
             params = _update_element_table_mapping_label_colors(sdata, params, wanted_elements_on_cs)
-        else:
+        if isinstance(params, (PointsRenderParams, ShapesRenderParams)):
             params = _validate_colors_element_table_mapping_points_shapes(sdata, params, wanted_elements_on_cs)
 
-    # if params.palette is None:
-    #     params.palette = [[None] for _ in wanted_elements_on_cs]
     image_flag = element_type == "images"
     return _match_length_elements_groups_palette(params, wanted_elements_on_cs, image=image_flag)
 
 
-def _is_coercable_to_float(series):
+def _is_coercable_to_float(series: pd.Series) -> bool:
     numeric_series = pd.to_numeric(series, errors="coerce")
     return not numeric_series.isnull().any()
+
+
+def _return_list_str_none(parameter: list[str | None] | str | None) -> list[str | None]:
+    """Force mypy to recognize list of string and None."""
+    if isinstance(parameter, list) and all(isinstance(item, (str, type(None))) for item in parameter):
+        checked_parameter = parameter if isinstance(parameter, list) else [None]
```

Comment:
You already checked whether it's a list?

Reply:
you can try remove it and see what mypy does:P

Reply:
ah yeah initially I had an issue with mypy not accepting it but that could have been due to another conflict. Now it passes when removing.